proposals/csharp-9.0/nullable-reference-types-specification.md

```diff
@@ -285,62 +307,154 @@ If a `simple_name` is not classified as a value, its null state is "not null". O
 
 If a `member_access` is not classified as a value, its null state is "not null". Otherwise, if it is a tracked expression, its null state is its tracked null state at this source location. Otherwise its null state is the default null state of its type.
 
+```c#
+var person = new Person();
+
+// The receiver is a tracked expression hence the member_access of the property 
+// is tracked as well 
+if (person.FirstName is not null)
+{
+    Use(person.FirstName);
+}
+
+// The return of an invocation is not a tracked expression hence the member_access
+// of the return is also not tracked
+if (GetPerson().FirstName is not null)
+{
+    Use(GetPerson().FirstName);
+}
+
+void Use(string s) 
+{ 
+    // ...
+}
+
+public class Person
+{
+    public string? FirstName { get; set; }
+    public string? LastName { get; set; }
+
+    private static Person s_anonymous = new Person();
+    public static Person GetAnonymous() => s_anonymous;
+}
+```
+
 ### Invocation expressions
 
 If an `invocation_expression` invokes a member that is declared with one or more attributes for special null behavior, the null state is determined by those attributes. Otherwise the null state of the expression is the default null state of its type.
 
+The null state of an `invocation_expression` is not tracked by the compiler.
+
+```c#
+
+// The result of an invocation_expression is not tracked
+if (GetText() is not null)
+{
+    // Warning: Converting null literal or possible null value to non-nullable type.
+    string s = GetText();
+    // Warning: Dereference of a possibly null reference.
+    Use(s);
+}
+
+// Nullable friendly pattern
```

Comment:
Not sure what is the purpose/intent for this example. The TODO list includes having a section on null and pure null tests. Patterns could be covered there.

Reply:
The intent of this example is to demonstarte that method invocations are different than properties / fields in that their results are not tracked.

Reply:
I think that's already achieved by the previous sample (under `// The result of an invocation_expression is not tracked`)

---
In reply to: [521708462](https://github.com/dotnet/csharplang/pull/4126#discussion_r521708462) [](ancestors = 521708462)

Reply:
Agreed but I wanted to try and keep every section as self contained as possible. Wanted to basically have supporting examples, when relevant, for the relevant sections.